app/models/user.rb

```diff
@@ -8,4 +8,5 @@ class User < ApplicationRecord
   has_many :rooms, dependent: :restrict_with_exception
 
   has_many :activities, dependent: :destroy
+  has_many :notifications, dependent: :destroy
```

Comment:
you need another relation one for being an actor (since there are two relations in Notification pointing to user)

Reply:
I just got rid of `:actor` per your feedback above